Assets/Tests/InputSystem/CoreTests_Editor.cs

```diff
@@ -834,36 +831,6 @@ public void Editor_InputActions_AssetFileUpdatedAfterMoveOrCopy(string newAssetP
         }
     }
 
-    [Test]
-    [Category("Editor")]
-    public void Editor_InputActionAssetManager_CanMoveAssetOnDisk()
-    {
-        const string kAssetPath = "Assets/DirectoryBeforeRename/InputAsset." + InputActionAsset.Extension;
-        const string kAssetPathAfterMove = "Assets/DirectoryAfterRename/InputAsset." + InputActionAsset.Extension;
-        const string kDefaultContents = "{}";
-
-        AssetDatabase.CreateFolder("Assets", "DirectoryBeforeRename");
-        File.WriteAllText(kAssetPath, kDefaultContents);
-        AssetDatabase.ImportAsset(kAssetPath);
-
-        var asset = AssetDatabase.LoadAssetAtPath<InputActionAsset>(kAssetPath);
-        Assert.NotNull(asset, "Could not load asset: " + kAssetPath);
-
-        var inputActionAssetManager = new InputActionAssetManager(asset);
-        inputActionAssetManager.Initialize();
-        inputActionAssetManager.onDirtyChanged = (bool dirty) => {};
```

Comment:
Yes, it was an odd design I modified since it didn't make sense to me and provided a somewhat messy API. Basically when this type is constructed via a constructor it doesn't make sense to require a call to Initialize after. This is now done as part of the constructor which is cleaner in my opinion. Initialize() remains available for serialisation scenarios when it should be called, but then no constructor is called.
The other odd thing here was that SaveChangesToAsset() would suprisingly throw if onDirtyChanged callback was not set. This didn't sit well I thought so I made it optional to set a callback and if you have registered one you get a callback on Save (transition to false), while its also fine to save without a callback.